resilience4j-bulkhead/src/test/java/io/github/resilience4j/bulkhead/BulkheadFutureTest.java

```diff
@@ -0,0 +1,212 @@
+package io.github.resilience4j.bulkhead;
+
+import io.github.resilience4j.test.HelloWorldService;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.*;
+import java.util.function.Supplier;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+
+public class BulkheadFutureTest {
+
+    private HelloWorldService helloWorldService;
+    private Future future;
+    private BulkheadConfig config;
+
+    @Before
+    public void setUp() {
+        helloWorldService = mock(HelloWorldService.class);
+        future = mock(Future.class);
+        config = BulkheadConfig.custom()
+                .maxConcurrentCalls(1)
+                .build();
+    }
+
+    @Test
+    public void shouldDecorateSupplierAndReturnWithSuccess() throws Exception {
+        Bulkhead bulkhead = Bulkhead.of("test", config);
```

Comment:
I think it would be better to use a Bulkhead mock in every test  and verify that `onComplete` is invoked.